projects/planner/src/app/diff/waypoints/diff-waypoints.component.ts

```diff
@@ -0,0 +1,81 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {TestData} from '../diff.component';
+import {faTasks} from '@fortawesome/free-solid-svg-icons';
+import {WayPoint} from '../../shared/models';
+import {UnitConversion} from '../../shared/UnitConversion';
+
+@Component({
+    selector: 'app-diff-waypoints',
+    templateUrl: './diff-waypoints.component.html',
+    styleUrls: ['./diff-waypoints.component.scss']
+})
+export class WaypointsDifferenceComponent implements OnInit {
+    @Input({required: true}) data!: TestData;
+    public tasks = faTasks;
+    public waypointRows: WaypointsTableRow[] = [];
+    constructor(
+        public units: UnitConversion
+    ) {
+    }
+
+    ngOnInit(): void {
+        const MAX_SAFETY_LIMIT = 65536; // 2**16
```

Comment:
Keep component empty. It should contain only logic related to rendering.
Introduce new service, which will prepare data for this component and make it testable.
See e.g. https://github.com/jirkapok/GasPlanner/blob/master/projects/planner/src/app/shared/waypoints.service.spec.ts

Reply:
For now simply extracted to a Provider class. Will rework into injectable service once importing testing profiles into DiveSchedule is implemented.